keps/sig-network/2079-network-policy-port-range/README.md

```diff
@@ -0,0 +1,296 @@
+
+* **How can an operator determine if the feature is in use by workloads?**
+  
+  Operators can determine if NetworkPolicies are making use of Range creating
+  an object specifying the range and validating if the traffic is allowed within 
+  the specified range
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  Operators would need to monitor the traffic of the Pods to verify if a 
+  specified port range is applied and allowed in their workloads
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+ N/A
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+ N/A
+
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+  No
+
+
+### Scalability
+
+_For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them._
+
+_For beta, this section is required: reviewers must answer these questions._
+
+_For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field._
+
+* **Will enabling / using this feature result in any new API calls?**
+  TBD
+
+* **Will enabling / using this feature result in introducing new API types?**
+  No, unless the new ``NetworkPolicyPortRange`` is considered a new API type
+
+* **Will enabling / using this feature result in any new calls to the cloud 
+provider?**
+  No
+
+* **Will enabling / using this feature result in increasing size or count of 
+the existing API objects?**
+
+  - API type(s): NetworkPolicy / NetworkPolicyPorts
+  - Estimated increase in size: New struct inside the object with two fields of 
+  16 bits each
+  - Estimated amount of new objects: N/A
+
+* **Will enabling / using this feature result in increasing time taken by any 
+operations covered by [existing SLIs/SLOs]?**
+  N/A
+
+* **Will enabling / using this feature result in non-negligible increase of 
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+  It might get some increase of resource usage by the CNI while parsing the 
+  new field.
+
+### Troubleshooting
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+  As this feature is mainly used by CNI providers, the reaction with API server 
+  and/or etcd being unavailable will be the same as before.
+
+* **What are other known failure modes?**
+  N/A
+
+* **What steps should be taken if SLOs are not being met to determine the problem?**
+  N/A
+
+## Implementation History
+- 2020-10-08 Initial [KEP PR](https://github.com/kubernetes/enhancements/pull/2079)
+
+## Drawbacks
+
+*  The technology used by the CNI provider might not support port range in a 
+trivial way. As an example, OpenFlow does not have a way to specify port range
```

Comment:
actually we can leverage the tcp_dst field in ovs in the port/mask format to specify range of ports and reduce the number of flows required. This is documented at https://www.openvswitch.org/support/dist-docs-2.5/ovs-ofctl.8.txt  | grep tcp_dst

Excerpt from the doc:
/*
              Bitwise  matches  on transport ports are rarely useful in isola‐
              tion, but a group of them can be used to reduce  the  number  of
              flows  required  to  match  on  a range of transport ports.  For
              example, suppose that the goal is to match TCP source ports 1000
              to  1999,  inclusive.   One way is to insert 1000 flows, each of
              which matches on a single source port.  Another way is  to  look
              at the binary representations of 1000 and 1999, as follows:
*/